libraries/resource_providers/api500/c7000/storage_pool_provider.rb

```diff
@@ -61,6 +61,7 @@ def remove
         def update
           load_storage_system
           desired_state = Marshal.load(Marshal.dump(@item.data))
+          desired_state.delete('type')
```

Comment:
any reason why are we deleting type key value pair?
I hope while retrieving the existing resource, type should be compared isn't it?

Reply:
The API docs says we dont need type field while doing update and it complains if we have that in our JSON body